source/com.microsoft.tfs.client.eclipse.ui.egit/src/com/microsoft/tfs/client/eclipse/ui/egit/teamexplorer/TeamExplorerProtocolHandlerNavigationLink.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the repository root.
+
+package com.microsoft.tfs.client.eclipse.ui.egit.teamexplorer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Shell;
+
+import com.microsoft.tfs.client.common.git.utils.GitHelpers;
+import com.microsoft.tfs.client.common.ui.teamexplorer.TeamExplorerContext;
+import com.microsoft.tfs.client.common.ui.teamexplorer.TeamExplorerNavigator;
+import com.microsoft.tfs.client.common.ui.teamexplorer.internal.TeamExplorerHelpers;
+import com.microsoft.tfs.client.common.ui.teamexplorer.internal.TeamExplorerNavigationItemConfig;
+import com.microsoft.tfs.client.common.ui.teamexplorer.link.TeamExplorerBaseNavigationLink;
+import com.microsoft.tfs.client.eclipse.ui.egit.Messages;
+import com.microsoft.tfs.client.eclipse.ui.egit.protocolhandler.ProtocolHandlerHelpers;
+
+public class TeamExplorerProtocolHandlerNavigationLink extends TeamExplorerBaseNavigationLink {
+
+    private static final Log log = LogFactory.getLog(TeamExplorerProtocolHandlerNavigationLink.class);
+
+    @Override
+    public boolean isEnabled(final TeamExplorerContext context) {
+        try {
+            return TeamExplorerHelpers.hasProtocolHandlerRequest();
+        } catch (final Exception e) {
+            log.error("", e); //$NON-NLS-1$
```

Comment:
This logging is for us as developers. No exception should appear in the production code. And the exception itself contains all information we need.